### PR TITLE
Page params parsing fix

### PR DESCRIPTION
**Pull request by lucacorti**
Fix for #191, also adjusts test.

### CODE REVIEW

File: test/jsonapi/plugs/query_parser_test.exs [hunk 1]
```diff
@@ -132,11 +132,11 @@ defmodule JSONAPI.QueryParserTest do
...
-    assert parse_pagination(config, %{"limit" => 1}).page.limit == 1
-    assert parse_pagination(config, %{"offset" => 1}).page.offset == 1
-    assert parse_pagination(config, %{"page" => 1}).page.page == 1
-    assert parse_pagination(config, %{"size" => 1}).page.size == 1
-    assert parse_pagination(config, %{"cursor" => 1}).page.cursor == 1
+    assert parse_pagination(config, %{"limit" => "1"}).page.limit == 1
+    assert parse_pagination(config, %{"offset" => "1"}).page.offset == 1
+    assert parse_pagination(config, %{"page" => "1"}).page.page == 1
+    assert parse_pagination(config, %{"size" => "1"}).page.size == 1
+    assert parse_pagination(config, %{"cursor" => "1"}).page.cursor == 1
```
Comment (jherdman) [inline test/jsonapi/plugs/query_parser_test.exs]: Does it make sense to add a case like `parse_pagination(config, %{"page" => "boogers"})`? This case would expose `String.to_integer/1` throwing when the String doesn't represent an Integer.

File: lib/jsonapi/plugs/query_parser.ex [hunk 2]
```diff
@@ -103,8 +103,23 @@ defmodule JSONAPI.QueryParser do
 
   def parse_pagination(config, map) when map_size(map) == 0, do: config
 
-  def parse_pagination(%Config{} = config, page),
-    do: Map.put(config, :page, struct_from_map(page, %Page{}))
+  def parse_pagination(%Config{} = config, page) do
```
[the remark this answers is not in the document]
Comment (doomspork) [reply lib/jsonapi/plugs/query_parser.ex]: In addition to a typespec, I think we can find a way to refactor this function to be a little easier to grok. What do you think?

**Comment by jherdman** [comment]
A thought occurred to me this weekend: what if the real bug is the typespec for the `Page`, and not this? Theoretically you can page using any sort of data. A theoretical calendar application, for instance, could page by time windows (e.g. weeks, days, months).

**Comment by lucacorti** [comment]
I think the issue here is JSON:API does not mandate any specific pagination scheme or interface beyond the usage of the `page` query param. So I think modeling a Page struct and mandating a specific shape on it is really something that could be avoided altogether.

Another approach to this would be ditching the `JSONAPI.Page` struct altogether, and just passing the `page` query param content to the pagination logic.

File: test/jsonapi/plugs/query_parser_test.exs (hunk 1 above)
Comment (lucacorti) [reply test/jsonapi/plugs/query_parser_test.exs]: Yes, also, cursor may rightfully be a string, not an integer.

File: lib/jsonapi/plugs/query_parser.ex [hunk 3]
```diff
@@ -103,8 +103,23 @@ defmodule JSONAPI.QueryParser do
...
+    page =
+      ["page", "size", "offset", "limit", "cursor"]
+      |> Enum.reduce(%{}, fn param, acc ->
+        param_name = Application.get_env(:jsonapi, String.to_atom("#{param}_query_param"), param)
```
[the remark this answers is not in the document]
Comment (lucacorti) [reply lib/jsonapi/plugs/query_parser.ex]: Yes, they come from there. Onestly, I'm proposing to get rid of the Page struct altogether in favor of directyle using query params, so I'm dropping this PR in favour of #189

**Pull request closed without merging**